File: frontend/src/app/domains/course/dtos/course-detail.dto.ts
import { UserSummaryDto } from '@/domains/user/dtos/user-summary.dto';

export interface CourseDetailDto {
  id: number;
  title: string;
  description: string;
  teacher: UserSummaryDto;
  price: number;
  imageUrl: string;
  level: string;
  durationHours: number;
  lessonCount: number;
}

File: backend/src/main/java/com/course/platform/course/dto/CourseDetailDto.java
package com.course.platform.course.dto;

import java.math.BigDecimal;

import com.course.platform.user.dto.UserSummaryDto;

import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CourseDetailDto {
    private Integer id;
    private String title;
    private String description;
    private UserSummaryDto teacher;
    private BigDecimal price;
}
